paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/utils/PaymentSheetTestRunner.kt

```diff
@@ -0,0 +1,85 @@
+package com.stripe.android.paymentsheet.utils
+
+import androidx.activity.ComponentActivity
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ActivityScenario
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.paymentsheet.CreateIntentCallback
+import com.stripe.android.paymentsheet.ExperimentalPaymentSheetDecouplingApi
+import com.stripe.android.paymentsheet.MainActivity
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.PaymentSheetResultCallback
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+internal class PaymentSheetTestRunnerContext(
+    private val scenario: ActivityScenario<MainActivity>,
+    private val paymentSheet: PaymentSheet,
+    private val countDownLatch: CountDownLatch,
+) {
+
+    fun presentPaymentSheet(
+        block: PaymentSheet.() -> Unit,
+    ) {
+        scenario.onActivity {
+            paymentSheet.block()
+        }
+    }
+
+    fun markTestSucceeded() {
```

Comment:
Is it worth adding a comment on when this is used?

Normally we know a test succeeds when it calls the result, but some tests don't expect the result to be called, so we mark it successful manually.